rust-mode.el

```diff
@@ -573,6 +584,27 @@ the desired identifiers), but does not match type annotations \"foo::<\"."
             ((not (looking-at (rx (0+ space) "<")))
 	     (throw 'rust-path-font-lock-matcher match))))))))
 
+(defvar rust-string-interpolation-re
+  "\\({[^}]*}\\)"
+  "Regular expression to match interpolation syntax in strings.")
+
+(defun rust-string-interpolation-matcher (limit)
+  (unless (rust-in-cmnt)
+    (save-match-data
+      ;; we presently ignore escaping closing delimiters
```

Comment:
Thanks, those gave me some useful ideas. Is nesting something that Rust implements? I thought it was only escaping.